drivers/samsung/displays/mdc_protocol_spec.cr

```diff
@@ -0,0 +1,34 @@
+  # [header, command, id, data.size, [data], checksum]
+
+DriverSpecs.mock_driver "Samsung::Displays::MDCProtocol" do
+  id = "\x00"
+
+  # connected -> do_poll
+  # power? will take priority over status as status has priority = 0
+  # power? -> panel_mute
+  should_send("\xAA\xF9#{id}\x00\xF9")
+  responds("\xAA\xFF#{id}\x03A\xF9\x00\xFF")
+  status[:power].should eq(true)
+  # status
+  should_send("\xAA\x00#{id}\x00\x00")
+  responds("\xAA\xFF#{id}\x09A\x00\x01\x06\x00\x14\x00\x00\x00\xFF")
+  status[:hard_off].should eq(false)
+  # status[:power].should eq(true)
```

Comment:
Is there a reason this is commented out?

Reply:
This is because I think the spec runner doesn't preserve the previous device state set here https://github.com/PlaceOS/drivers/blob/59fe9aa630abac79aa710ade043602e70706652c/drivers/samsung/displays/mdc_protocol_spec.cr#L11

Reply:
actually it's I think because this line get's run if the setting exists even though `self[:hard_off] == false` https://github.com/PlaceOS/drivers/pull/19/commits/3781786578535012b59047db762ceec920412bb5#diff-96bf90ab842197ca263274d3fdb8e648L263

Reply:
made another commit which is probably more ideal and works with crystal better https://github.com/PlaceOS/drivers/pull/19/commits/95f64ba271b69c74ebc321d2e158dda0b42fcaf7